include/vpnes/core/bus.hpp

```diff
@@ -362,7 +362,7 @@ class CBus {
 	template <class T>
 	void addPreReadHook(std::uint16_t addr, T *device,
 	    typename CAddrHookMapped<T>::addrHook_t hook) {
-		m_ReadHooksPre.emplace(addr, new CAddrHookMapped<T>(device, hook));
+		m_ReadHooksPre.emplace(addr, std::unique_ptr<CAddrHook>(new CAddrHookMapped<T>(device, hook)));
```

Comment:
`std::is_constructable<std::unique_ptr<CAddrHook>, CAddrHookMapped<T> *>::value` should be `true`. Is it a compiler issue?